core/src/execution/datafusion/planner.rs

```diff
@@ -566,8 +567,23 @@ impl PhysicalPlanner {
                 Ok(Arc::new(NotExpr::new(child)))
             }
             ExprStruct::Negative(expr) => {
-                let child = self.create_expr(expr.child.as_ref().unwrap(), input_schema)?;
-                Ok(Arc::new(NegativeExpr::new(child)))
+                let child: Arc<dyn PhysicalExpr> =
+                    self.create_expr(expr.child.as_ref().unwrap(), input_schema.clone())?;
+                let eval_mode = match expr.eval_mode.as_str() {
+                    "ANSI" => EvalMode::Ansi,
+                    "TRY" => EvalMode::Try,
+                    "LEGACY" => EvalMode::Legacy,
+                    other => {
+                        return Err(ExecutionError::GeneralError(format!(
+                            "Invalid EvalMode: \"{other}\""
+                        )))
+                    }
+                };
```

Comment:
We can make use of the generated protobuf enum here rather than hard-code the strings. This would also allow us to remove the "other" error handling case.

```rust
                let eval_mode = match spark_expression::EvalMode::try_from(expr.eval_mode)? {
                    spark_expression::EvalMode::Ansi => EvalMode::Ansi,
                    spark_expression::EvalMode::Try => EvalMode::Try,
                    spark_expression::EvalMode::Legacy => EvalMode::Legacy,
                };
```

Reply:
This is nice but I don't see EvalMode in spark_expressions.

I took inspiration from here, Cast expression does similar match -  https://github.com/apache/datafusion-comet/pull/471/files#diff-264772aa6994f16ba436ff47d7d15a126e58e78a46d617f22107b376dfbd31e8L351

Reply:
Oh, my mistake, we won't have this until https://github.com/apache/datafusion-comet/pull/415 is merged.